uniro_frontend/src/components/customInput.tsx

```diff
@@ -1,35 +1,44 @@
-import { ChangeEvent, InputHTMLAttributes, useCallback, useState } from "react";
+import { ChangeEvent, InputHTMLAttributes, useCallback, useEffect, useRef, useState } from "react";
 import Search from "../../public/icons/search.svg?react";
 import ChevronLeft from "../../public/icons/chevron-left.svg?react";
 import Mic from "../../public/icons/mic.svg?react";
 import Close from "../../public/icons/close-circle.svg?react";
 
 interface CustomInputProps extends InputHTMLAttributes<HTMLInputElement> {
-	onLengthChange: (e: string) => void;
+	onChangeDebounce: (e: string) => void;
 	placeholder: string;
 	handleVoiceInput: () => void;
 }
 
-export default function Input({ onLengthChange, placeholder, handleVoiceInput, ...rest }: CustomInputProps) {
+export default function Input({ onChangeDebounce, placeholder, handleVoiceInput, ...rest }: CustomInputProps) {
 	const [isFocus, setIsFocus] = useState<boolean>(false);
-	const [tempValue, setTempValue] = useState<string>("");
-	const [currentValue, setCurrentValue] = useState<string>("");
+	const [value, setValue] = useState<string>("");
+	const timeOutRef = useRef<number>();
```

Comment:
number로 사용하게 되면, node.js 환경에서 실행했을 때, 리턴되는 타입이 NodeJS.Timeout이라, type 충돌이 날 수도 있다고 합니다! 브라우저 환경에서 문제는 없겠지만, 안전한 type을 위해서 다음 코드는 어떠신가요??

```suggestion
	const timeoutRef = useRef<ReturnType<typeof setTimeout> | null>(null);
```

Reply:
고려해서 리팩토링 진행해보겠습니다